Yafc/Windows/WelcomeScreen.cs

```diff
@@ -296,6 +308,17 @@ private void BuildPathSelect(ImGui gui, ref string path, string description, str
             }
 
             if (gui.RemainingRow(0f).BuildTextInput(path, out path, placeholder)) {
+                switch (editType) {
+                    case EditType.Workspace:
+                        this.path = path;
+                        break;
+                    case EditType.Factorio:
+                        dataPath = path;
+                        break;
+                    case EditType.Mods:
+                        modsPath = path;
+                        break;
+                }
```

Comment:
Could you please explain the addition of this switch? What problem does it address? I don't understand how it worked before if we started assigning these variables only now.

Reply:
The old code used `ref string path` as the parameter, so `BuildTextInput(..., out path, ...)` modified the field directly. When I changed `modsPath` to a property, I had to change to a non-ref parameter, and that interrupted the direct-to-field write.

Reply:
Good call. The fewer `ref` and `out` we have, the better.